tests/test_all_fields.py

```diff
@@ -205,24 +166,38 @@ def test_all_fields_for_streams_are_replicated(self):
                 # gather expectations
                 expected_all_keys = self.selected_fields.get(stream_name, set())
 
-                if self.synced_records.get(stream_name):
-                    # gather results
-                    actual_all_keys_per_record = [set(message['data'].keys()) for message in
-                                                  self.get_upsert_messages_for_stream(self.synced_records, stream_name)]
+                records = [
+                    record['data'] for record in
+                    self.synced_records.get(stream, {}).get('messages', [])
+                    if record.get('action') == 'upsert']
 
-                    for actual_all_keys in actual_all_keys_per_record:
-                        self.assertSetEqual(expected_all_keys, actual_all_keys)
+                if records:
+                    fields_replicated = self.actual_fields.get(stream, set())
 
+                    # verify that all fields are sent to the target
+                    # test the combination of all records
+                    self.assertSetEqual(
+                        fields_replicated, expected_all_keys,
+                        logging=f"verify all fields are replicated for stream {stream}")
 
     ##########################################################################
-    ### Tests To Skip
+    # Tests To Skip
     ##########################################################################
 
-
     @unittest.skip("Random selection doesn't always sync records")
     def test_all_streams_sync_records(self):
         pass
 
-
-    def __init__(self, test_run):
-        super().__init__(test_run)
+#  TODO - Run failed, might have an issue with the way we expect data for some fields
```

Comment:
Is this error consistent? This seems related to the `(other)` behavior that google does when a report is too large: https://support.google.com/analytics/answer/13331684?visit_id=638206372168248675-70649747&rd=2

Reply:
The tap writes a schema with anyOf string or datetime https://github.com/singer-io/tap-ga4/blob/main/tap_ga4/discover.py#L68-L75 
which should allow the `(other)` field to be replicated. The list doesn't include nth_hour so that could be the issue here.

Reply:
I don't know if the error is consistent because we randomly select fields and therefore aren't always selecting these. I certainly didn't see the error often.  But I also worried that for these fields we might have a real problem.